### DISCUSSION TITLE
I can't add a new field in the woocommerce checkout form

### DISCUSSION THREAD

**Discussion by s1moz23** [woocommerce/woocommerce#52594] [WooCommerce, Blocks, and Site Editing]
I need to add the DNI field to the WooCommerce payment form, when I try to add it in the WooCommerce "Payment Form" option it does not appear on the website attached image 1
![1](https://github.com/user-attachments/assets/3d051a38-611b-4303-a131-872a4a645c04)
. Then I try to edit it with the Wordpress editor and all the blocked options appear I attach image 2
![2](https://github.com/user-attachments/assets/90d5d96f-116e-4faf-9a17-7876f8c4dddc)
. I absolutely need to have that field to be able to invoice. Can you help me with this case?

Thanks

**Comment by nielslange** [comment]
Hello @s1moz23 👋

Thanks for reaching out. Based on your screenshot, it seems that you are using the [Checkout Field Editor](https://woocommerce.com/document/checkout-field-editor/) to add extra fields to the Billing Address section. Unfortunately, the Checkout Field Editor is currently not compatible with the Checkout block. However, you could use the [additional fields API](https://developer.woocommerce.com/docs/cart-and-checkout-additional-checkout-fields/) to add additional fields to the checkout.
